src/Symfony/Component/Finder/Iterator/LazyIterator.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Finder\Iterator;
+
+/**
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class LazyIterator implements \Iterator
+{
+    /** @var callable */
+    private $iteratorFactory;
+    /** @var \Iterator */
+    private $iterator;
+
+    public function __construct(callable $iteratorFactory)
+    {
+        $this->iteratorFactory = $iteratorFactory;
+    }
+
+    public function current()
+    {
+        return ($this->iterator ?? $this->iterator = ($this->iteratorFactory)())->current();
```

Comment:
if you call `current()` after the iterator has been unset (due to reaching the end), this would restart the iteration (same for all other methods btw)

Reply:
good catch!!!! thanks a lot.

Fixed and Test added